libraries/AP_InertialSensor/AP_InertialSensor_MPU6000.cpp

```diff
@@ -299,9 +299,22 @@ bool AP_InertialSensor_MPU6000::_init()
 
 void AP_InertialSensor_MPU6000::_fifo_reset()
 {
+    /* We read first the user control register status to
+       prevent a reset to I2C Master if it was enabled.
+       We need to do it here because it's obviously that FIFO was enabled
+       previously.
+    */
+    uint8_t user_ctrl = _register_read(MPUREG_USER_CTRL);
+
+    /* Reset the FIFO buffer */
     _register_write(MPUREG_USER_CTRL, 0);
     _register_write(MPUREG_USER_CTRL, BIT_USER_CTRL_FIFO_RESET);
     _register_write(MPUREG_USER_CTRL, BIT_USER_CTRL_FIFO_EN);
+
+    /* We come back from last user control register status and
+       continue the normal hardware operation.
+    */
+    _register_write(MPUREG_USER_CTRL, user_ctrl);
```

Comment:
I don't think this should be like this. There are configurations in this register that needs to be done as soon as the sensor is enabled. See `4.29 Register 106 – User Control USER_CTRL` in the datasheet. Particularly `I2C_MST
_EN` and `I2C_IF_DIS`. We actually need to leave these bits as they were before.

So... from reading the datasheet I **think** what we need to do here is to replace the lines above with:

```
user_ctrl &= ~BIT_USER_CTRL_FIFO_EN;
_register_write(MPUREG_USER_CTRL, user_ctrl);
_register_write(MPUREG_USER_CTRL, user_ctrl | BIT_USER_CTRL_FIFO_RESET);
_register_write(MPUREG_USER_CTRL, user_ctrl | BIT_USER_CTRL_FIFO_EN);
```

This needs to be properly tested and validated though.